internal/runners/clean/run.go

```diff
@@ -33,3 +38,16 @@ func removeCache(cachePath string) error {
 	}
 	return nil
 }
+
+func undoPrepare(out output.Outputer) {
+	toRemove := prepare.InstalledPreparedFiles()
+
+	for _, f := range toRemove {
+		if fileutils.TargetExists(f) {
+			err := os.Remove(f)
+			if err != nil {
+				out.Notice(locale.Tl("[ERROR]Warning: [/RESET] Could not remove file {{.V0}}.", f))
```

Comment:
I changed the entire behavior, because it was already inconsistent (sometimes we stop the un-installation half-way and sometimes we delete as much as possible).

Now: If an error occurs at any point, we are still trying to remove as much as possible.

We also report the files that could not be removed, such that the user can remove them manually.  IMO that appears to be a better approach than leaving the user with a half-way unusuble installation and no way to fix it or re-try it (in case the State Tool was already removed).

An even cleaner way would be functionality to roll back the un-installation if anything fails, and I filed a follow-up story for that: https://www.pivotaltracker.com/story/show/178134918